src/net/sourceforge/kolmafia/request/CharPaneRequest.java

```diff
@@ -292,7 +292,7 @@ public static final String getLastResponse() {
 
   public static final Pattern AVATAR_PATTERN =
       Pattern.compile(
-          "<img +src=[^>]*?(?:cloudfront.net|images.kingdomofloathing.com|/images)/([^>'\"\\s]+)");
+          "<img [ crossorigin=\"Anonymous\"|]+src=[^>]*?(?:cloudfront.net|images.kingdomofloathing.com|/images)/([^>'\"\\s]+)");
```

Comment:
square brackets? as in a character class?

Reply:
also, you've shifted the `+` from the space to your new construct.

You probably wanted
```
"<img +(crossorigin=\"Anonymous\")?src=[^>]*?(?:cloudfront.net|images.kingdomofloathing.com|/images)/([^>'\"\\s]+)");
```

Reply:
Also use "(?: ... )" to make it a non-capturing group so to not shift group numbers.

Reply:
Done